File: robot-server/src/main/java/com/informatica/servidor/logica/BrazoRobotFabrica.java
package com.informatica.servidor.logica;

import com.informatica.servidor.utils.Constantes;

/**
 *
 * @author Vamaya
 */
public class BrazoRobotFabrica {

    //CONSTRUCTOR
    public BrazoRobotFabrica() {
    }
    
    //METODOS
    public static Dedo crearDedo(){
        //Se inicializan los valores de la falange digital
        int xiFalDig = Constantes.ANCHO_MAXIMO_CANVAS/2;
        int yiFalDig = 0;
        int wFalDig = 10;
        int hFalDig = Constantes.ALTO_DEDO/2;

        //Se inicializan los valores de la falange proximal
        int xiFalProx = xiFalDig;
        int yiFalProx = yiFalDig + hFalDig;        
        int wFalProx = 10;
        int hFalProx = Constantes.ALTO_DEDO/2;
        
        return new Dedo(new Hueso(xiFalDig, yiFalDig, xiFalDig, yiFalDig+hFalDig, wFalDig), xiFalProx, yiFalProx, xiFalProx, yiFalProx+hFalProx, wFalProx);
    }
    
    
    
    public static Mano crearMano(Dedo dedoIzq, Dedo dedoDer){
        int xiMano = Constantes.ANCHO_MAXIMO_CANVAS/2;
        int yiMano = Constantes.ALTO_DEDO;
        int wMano = 20;
        int hMano = Constantes.ALTO_MANO;
        
        return new Mano(dedoIzq, dedoDer, xiMano, yiMano, xiMano, yiMano + hMano, wMano);
    
    }
    
    public static AnteBrazo crearAntebrazo(Mano mano) {
        //Se inicializan los valores del antebrazo
        int xiAnteBrazo = Constantes.ANCHO_MAXIMO_CANVAS/2;
        int yiAnteBrazo = Constantes.ALTO_DEDO+Constantes.ALTO_MANO;       
        int wAnteBrazo = 20;
        int hAnteBrazo = Constantes.ALTO_ANTEBRAZO;
        
        return new AnteBrazo(mano, xiAnteBrazo, yiAnteBrazo, xiAnteBrazo, yiAnteBrazo + hAnteBrazo, wAnteBrazo);
        
    }
    
    
    public static Brazo crearBrazo(AnteBrazo anteBrazo) {
        //Se inicializan los valores del brazo
        int xiBrazo = Constantes.ANCHO_MAXIMO_CANVAS/2;
        int yiBrazo = Constantes.ALTO_DEDO+Constantes.ALTO_MANO+Constantes.ALTO_ANTEBRAZO;
        int wBrazo = 20;
        int hBrazo = Constantes.ALTO_BRAZO;
        
        return new Brazo(anteBrazo, xiBrazo, yiBrazo, xiBrazo, yiBrazo + hBrazo, wBrazo);
    }
    
    
    public static Base crerBase(){
        //Se inicializan los valores del brazo
        int xiBrazo = Constantes.ANCHO_MAXIMO_CANVAS/2;
        int yiBrazo = Constantes.ALTO_DEDO+Constantes.ALTO_MANO+Constantes.ALTO_ANTEBRAZO;
        int wBrazo = 20;
        int hBrazo = Constantes.ALTO_BRAZO;
        return new Base(xiBrazo, yiBrazo+hBrazo, 100, wBrazo/2);
    }
    
    
}

File: my-app/src/logica/BrazoRobotFabrica.ts
import { Dedo } from './Dedo';
import { Hueso } from './Hueso';
import { Mano } from './Mano';
import { AnteBrazo } from './AnteBrazo';
import { Brazo } from './Brazo';
import { Base } from './Base';
import { Constantes } from 'src/utils/Constantes';

export class BrazoRobotFabrica {

    constantes: Constantes;

    constructor() {
        this.constantes = new Constantes();
    }

    //METODOS
    crearDedo(): Dedo {
        //Se inicializan los valores de la falange digital
        let xiFalDig: number = this.constantes.ANCHO_MAXIMO_CANVAS / 2;
        let yiFalDig: number = 0;
        let wFalDig: number = 10;
        let hFalDig: number = this.constantes.ALTO_DEDO / 2;

        //Se inicializan los valores de la falange proximal
        let xiFalProx: number = xiFalDig;
        let yiFalProx: number = yiFalDig + hFalDig;
        let wFalProx: number = 10;
        let hFalProx: number = this.constantes.ALTO_DEDO / 2;

        return new Dedo(new Hueso(xiFalDig, yiFalDig, xiFalDig, yiFalDig + hFalDig, wFalDig),
            xiFalProx, yiFalProx, xiFalProx, yiFalProx + hFalProx, wFalProx);
    }

    crearMano(dedoIzq: Dedo, dedoDer: Dedo): Mano {
        let xiMano: number = this.constantes.ANCHO_MAXIMO_CANVAS / 2;
        let yiMano: number = this.constantes.ALTO_DEDO;
        let wMano: number = 20;
        let hMano: number = this.constantes.ALTO_MANO;

        return new Mano(dedoIzq, dedoDer, xiMano, yiMano, xiMano, yiMano + hMano, wMano);
    }

    crearAntebrazo(mano: Mano): AnteBrazo {
        //Se inicializan los valores del antebrazo
        let xiAnteBrazo: number = this.constantes.ANCHO_MAXIMO_CANVAS / 2;
        let yiAnteBrazo: number = this.constantes.ALTO_DEDO + this.constantes.ALTO_MANO;
        let wAnteBrazo: number = 20;
        let hAnteBrazo: number = this.constantes.ALTO_ANTEBRAZO;

        return new AnteBrazo(mano, xiAnteBrazo, yiAnteBrazo, xiAnteBrazo, yiAnteBrazo + hAnteBrazo, wAnteBrazo);
    }

    crearBrazo(anteBrazo: AnteBrazo): Brazo {
        //Se inicializan los valores del brazo
        let xiBrazo: number = this.constantes.ANCHO_MAXIMO_CANVAS / 2;
        let yiBrazo: number = this.constantes.ALTO_DEDO + this.constantes.ALTO_MANO + this.constantes.ALTO_ANTEBRAZO;
        let wBrazo: number = 20;
        let hBrazo: number = this.constantes.ALTO_BRAZO;

        return new Brazo(anteBrazo, xiBrazo, yiBrazo, xiBrazo, yiBrazo + hBrazo, wBrazo);
    }

    crerBase(): Base {
        //Se inicializan los valores del brazo
        let xiBrazo: number = this.constantes.ANCHO_MAXIMO_CANVAS / 2;
        let yiBrazo: number = this.constantes.ALTO_DEDO + this.constantes.ALTO_MANO + this.constantes.ALTO_ANTEBRAZO;
        let wBrazo: number = 20;
        let hBrazo: number = this.constantes.ALTO_BRAZO;
        return new Base(xiBrazo, yiBrazo + hBrazo, 100, wBrazo / 2);
    }
}
